spring-kafka/src/main/java/org/springframework/kafka/annotation/RetryableTopicAnnotationProcessor.java

```diff
@@ -173,9 +179,13 @@ private SleepingBackOffPolicy<?> createBackoffFromAnnotation(Backoff backoff, Be
 			if (backoff.random()) {
 				policy = new ExponentialRandomBackOffPolicy();
 			}
-			policy.setInitialInterval(min);
+			if (min != null) {
+				policy.setInitialInterval(min);
+			}
 			policy.setMultiplier(multiplier);
-			policy.setMaxInterval(max > min ? max : ExponentialBackOffPolicy.DEFAULT_MAX_INTERVAL);
+			if (max != null && min != null && max > min) {
```

Comment:
What will' max' be if `max <= min`? Currently, we set it to `ExponentialBackOffPolicy.DEFAULT_MAX_INTERVAL` in that case. The new code seems to change that semantics.

Reply:
Good comment! Below is the relevant code snippet:
```
public class ExponentialBackOffPolicy implements SleepingBackOffPolicy<ExponentialBackOffPolicy> {
        ... ...
	/**
	 * The maximum value of the backoff period in milliseconds.
	 */
	private long maxInterval = DEFAULT_MAX_INTERVAL;
```
so regardless of whether we set the same default value or not in `RetryableTopicAnnotationProcessor` above, the end result is the same. I avoided duplication to make it more elegant.

Reply:
Let me know whether we need to revert back old behaviour. I am perfectly fine with either way.

Reply:
But don't we need to set that value in `RetryableTopicAnnotationProcessor`? Otherwise, how does this `max` variable get to that value? I am fine with cleaning it up if it is redundant. I am trying to see the connection.

Reply:
As I showed above, the policy class will set the fields to default values already just after the instance was initiated but before constructor is invoked, so after the default constructor was invoked, we have ended up with valid state already. 

Am I missing something here?

Reply:
Ok, I got it. Thanks for the explanation.